gpytorch/lazy/constant_mul_lazy_tensor.py

```diff
@@ -153,3 +155,11 @@ def diag(self):
     def evaluate(self):
         res = self.base_lazy_tensor.evaluate()
         return res * self.expanded_constant
+
+    @cached(name="root_decomposition")
+    def root_decomposition(self, method: Optional[str] = None):
+        if (self._constant > 0).sum() == self._constant.numel():
```

Comment:
Not sure how much `>=` messes with p.d. assumptions anywhere, but should be ok 

```suggestion
        if torch.all(self._constant >= 0):
```